cmd/clusterctl/client/common_test.go

```diff
@@ -51,15 +57,14 @@ func Test_parseProviderName(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			gotName, gotVersion, err := parseProviderName(tt.args.provider)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
-			if gotName != tt.wantName {
-				t.Errorf("gotName = %v, want %v", gotName, tt.wantName)
-			}
-			if gotVersion != tt.wantVersion {
-				t.Errorf("gotVersion = %v, want %v", gotVersion, tt.wantVersion)
+			if tt.wantErr {
```

Comment:
nit we should be consistent in having assertion on the operation err/on wantErr like
```
if tt.wantErr {
   g.Expect(err).To(HaveOccurred())
   return
}				
g.Expect(err).NotTo(HaveOccurred())

```
Without empty line in the middle and with an empty line after